core/src/storage/cell/sync_cell.rs

```diff
@@ -305,12 +305,15 @@ impl<T> parity_codec::Decode for SyncCell<T> {
 
 impl<T> Flush for SyncCell<T>
 where
-    T: parity_codec::Encode,
+    T: parity_codec::Encode + Flush,
 {
     fn flush(&mut self) {
         if self.cache.is_dirty() {
-            match self.cache.get() {
-                Some(val) => self.cell.store(val),
+            match self.cache.get_mut() {
+                Some(val) => {
+					self.cell.store(val);
+					val.flush();
+				},
```

Comment:
Formatting: mixed tabs and spaces here

Reply:
I probably need to run rustfmt manually.